src/system-health/health_checker/service_checker.py

```diff
@@ -24,23 +43,171 @@ class ServiceChecker(HealthChecker):
 
     def __init__(self):
         HealthChecker.__init__(self)
+        self.critical_process_dict = {}
+        # Containers that has invalid critical_processes file
+        self.bad_containers = set()
+
+        self.container_feature_dict = {}
+
+        self.need_save_cache = False
+
+        self.load_critical_process_cache()
+
+    def get_expected_runnning_container_set(self, feature_table):
+        """Get a set of containers that are expected to running on SONiC
+
+        Args:
+            feature_table (object): FEATURE table in CONFIG_DB
+
+        Returns:
+            set: A set of container names
+        """
+        containers = set()
```

Comment:
Do we consider renaming this variable containers to expected_running_containers?